agent/tests/Connection.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Tests;
+
+use Evenement\EventEmitter;
+use React\Socket\ConnectionInterface;
+use React\Stream\WritableStreamInterface;
+use RuntimeException;
+
+class Connection extends EventEmitter implements ConnectionInterface
+{
+    public function getRemoteAddress()
+    {
+        throw new RuntimeException(__FUNCTION__);
+    }
+
+    public function getLocalAddress()
+    {
+        throw new RuntimeException(__FUNCTION__);
+    }
+
+    public function isReadable()
+    {
+        throw new RuntimeException(__FUNCTION__);
+    }
+
+    public function pause()
+    {
+        throw new RuntimeException(__FUNCTION__);
+    }
+
+    public function resume()
+    {
+        throw new RuntimeException(__FUNCTION__);
+    }
+
+    /**
+     * @param  array<mixed>  $options
+     */
+    public function pipe(WritableStreamInterface $dest, array $options = [])
+    {
+        throw new RuntimeException(__FUNCTION__);
+    }
+
+    public function close()
+    {
+        //
+    }
```

Comment:
We call this method in the server. We don't _yet_ need it to do anything in our tests. Eventually we will mark the connection is closed to make assertions against it.